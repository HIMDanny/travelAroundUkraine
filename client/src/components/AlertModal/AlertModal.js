import {
  Button as MuiButton,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  styled,
} from '@mui/material';
import React from 'react';
import PropTypes from 'prop-types';

const Button = styled(MuiButton)({
  padding: '12px 24px',
});

const AlertModal = ({ open, onClose, onSubmit, title, children, disableCancelButton, submitButtonText, success }) => (
  <Dialog
    open={open}
    onClose={onClose}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
  >
    <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
    <DialogContent>
      <DialogContentText id="alert-dialog-description">{children}</DialogContentText>
    </DialogContent>
    <DialogActions sx={{ paddingBottom: 2 }}>
      {!disableCancelButton && (
        <Button onClick={onClose} disableElevation>
          Cancel
        </Button>
      )}

      <Button onClick={onSubmit} success={success ? 'true' : null} disableElevation autoFocus>
        {submitButtonText}
      </Button>
    </DialogActions>
  </Dialog>
);

AlertModal.propTypes = {
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  disableCancelButton: PropTypes.bool,
  submitButtonText: PropTypes.string,
  success: PropTypes.bool,
};

AlertModal.defaultProps = {
  title: 'Alert!',
  disableCancelButton: false,
  submitButtonText: 'Submit',
  success: false,
};

export default AlertModal;
