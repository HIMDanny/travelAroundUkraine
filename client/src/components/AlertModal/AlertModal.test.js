/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import AlertModal from './AlertModal';

jest.mock('@mui/material/Button', () => ({ children }) => <button type="button">{children}</button>);
jest.mock('@mui/material/Dialog', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogActions', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogContent', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/DialogContentText', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogTitle', () => ({ children }) => <h3>{children}</h3>);

describe('AlertModal snapshot test', () => {
  test('should AlertModal match snapshot', () => {
    const onClose = jest.fn();
    const onSubmit = jest.fn();

    const { asFragment } = render(
      <AlertModal open onClose={onClose} onSubmit={onSubmit} title="Alert modal">
        It is a text content of alert modal
      </AlertModal>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
