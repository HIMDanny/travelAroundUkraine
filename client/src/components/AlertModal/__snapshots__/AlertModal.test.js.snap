// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal snapshot test should AlertModal match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Alert modal
    </h3>
    <div>
      <div>
        It is a text content of alert modal
      </div>
    </div>
    <div>
      <button
        type="button"
      >
        Cancel
      </button>
      <button
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</DocumentFragment>
`;
